python/test/test_loops.py

```diff
@@ -34,39 +35,47 @@
 ]
 
 
+@pytest.fixture
+def lock():
+    return Lock()
+
+
 @pytest.mark.parametrize("modes", simple_modes)
-def test_do_all_python(modes):
+def test_do_all_python(modes, lock):
```

Comment:
From the little that I have read my understanding is that a new "Lock" instance would be passed per call? 

https://docs.pytest.org/en/6.2.x/fixture.html#requesting-fixtures

Not too familiar with pytest but just wanted to make sure each thread is accessing the same lock instance.

Reply:
ahh nvm actually i see the method f() called inside this method in parallel